### PR TITLE
Features/ft 1237 user friendlier analysis result download

### PR DESCRIPTION
**Pull request by forus**

### CODE REVIEW

File: src/groovy/jobs/steps/SimpleDumpTableResultStep.groovy [hunk 1]
```diff
@@ -34,7 +36,11 @@ class SimpleDumpTableResultStep extends AbstractDumpStep {
     }
 
     void writeHeader(CSVWriter writer) {
-        writer.writeNext(headers as String[])
+        List<String> fileHeaders = headers
+        if (headerMessageClosure) {
+            fileHeaders = headers.collect { headerMessageClosure(it) }
+        }
+        writer.writeNext(fileHeaders as String[])
```
Comment (cataphract) [inline src/groovy/jobs/steps/SimpleDumpTableResultStep.groovy]: This doesn't make sense to me. You already heave a protected `getHeaders` method. Why would you on top of that provide in the same class a method to transform those headers?

**Comment by cataphract** [comment]
I'll copy here the comments on HipChat.

The header rewriting seems fairly overcomplicated the way it's done now. It was already somewhat difficult to track the header names before. You have the `header` property `ColumnConfigurator` (possibly nested inside other), then this header moves inside the `Column`s in `Table`, and then the dump step itself could use these headers or not or transform them in some way (e.g. divide columns into two or three, creating new headers in the process).

This PR further complicates this, as there are two extra rewrites: one on the job level, and another at the dump level.

`AbstractAnalysisJob::headerMessage` should definitely not exist. It's a protected method that's not used by `AbstractAnalysisJob` at all. It's only used on a subclass, `CategoricalOrBinnedJob`, where it's transformed into a closure and written to `SimpleDumpTableResultStep::headerMessageClosure`. Which brings as to the second problem. `SimpleDumpTableResultStep` already has this method:

    protected List<String> getHeaders() {
        table.headers
    }

So it has a protected method to retrieve the headers, and a closure to transform these headers. Seems also unnecessary. Either a protected method (to customize via subclassing) or just the closure (to customize via property assignment) should be enough.

**Comment by timdo** [comment]
@forus While doing another round of testing with Rianne, I noticed that I missed one nasty side-effect of R-side code change, which broke GUI parsing of result-files. I'll add a ticket for it for next sprint, but this needs to be fixed before merge is possible. I'm assuming that due to Gustavo's comments this merge will not happen anytime soon anyway.

File: src/groovy/jobs/AbstractLowDimensionalAnalysisJob.groovy [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import jobs.steps.MultiRowAsGroupDumpTableResultsStep
+import jobs.steps.Step
+import jobs.steps.WriteFileStep
+import jobs.steps.helpers.ColumnConfigurator
+import jobs.table.Table
+import org.springframework.beans.factory.annotation.Autowired
+
+import static jobs.steps.AbstractDumpStep.getDEFAULT_OUTPUT_FILE_NAME
+
+abstract class AbstractLowDimensionalAnalysisJob extends AbstractAnalysisJob {
```
Comment (cataphract) [inline src/groovy/jobs/AbstractLowDimensionalAnalysisJob.groovy]: Why `AbstractLowDimensionalAnalysisJob`? It seems this is used by analyses that take high dimensional data, like the scatter plot.

**Pull request closed without merging**